beaker/client/application_client.py

```diff
@@ -64,12 +64,12 @@ def __init__(
 
     def compile(
         self, teal: str, source_map: bool = False
-    ) -> tuple[bytes, str, SourceMap]:
+    ) -> tuple[bytes, str, Optional[SourceMap]]:
         result = self.client.compile(teal, source_map=source_map)
```

Comment:
I wonder if, instead of casting this below, since this API is also used in `precompile.py`, if it's worth creating a type for this result?

Like so:

```py
from typing import TypedDict, Any

class AlgodCompileResponseOptionals(TypedDict, total=False):
    sourcemap: dict[str, Any]
    """JSON of the source map"""


class AlgodCompileResponse(AlgodCompileResponseOptionals):
    """Successful result of POST /v2/teal/compile

    ref: https://developer.algorand.org/docs/rest-apis/algod/v2/#tealcompile-response-200
    """

    hash: str
    """base32 SHA512_256 of program bytes (Address style)"""
    result: str
    """base64 encoded program bytes"""
```

(the weird thing of requiring two classes is because `typing.NotRequired` is only available in 3.11)

Reply:
I like the extra typing but the extra cast doesn't offend me too much. How strongly do you feel about this?

Reply:
I think it would be best if AlgoSDK exposed these types, so I don't feel too strongly about the `cast` to work around that fact.

Reply:
lets wait on this specific one and leave the cast for now